### PR TITLE
Scfdot: added script

### PR DESCRIPTION
**Pull request by grueni**
The makefile of scfdot contains some tasks with scfdot, dot and awk to create a graph with a legend.
I created a script with this commands and added ghostscript to create a second output file in pdf format.

### CODE REVIEW

File: components/sysutils/scfdot/manifests/sample-manifest.p5m [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright 2016 <contributor>
```
Comment (alarcher) [inline components/sysutils/scfdot/manifests/sample-manifest.p5m]: Do not commit this file since nothing has changed aside from the year.

**Review by alarcher — CHANGES_REQUESTED** [review]
Could you bump the component revision since a file was added?

**Review by alarcher — APPROVED** [review]

**Review by xen0l — COMMENTED** [review]
Output from shellcheck:
```

In scfdotctl line 19:
echo "-> Ups! We need graphviz and ghostscript, please install the packages with:"
     ^-- SC2039: In POSIX sh, echo flags are undefined.


In scfdotctl line 162:
scfdot ${SCFDOTOPTSH}     > /tmp/hostname.dot
       ^-- SC2086: Double quote to prevent globbing and word splitting.


In scfdotctl line 168:
mv hostname.ps  `hostname`.ps
                ^-- SC2046: Quote this to prevent word splitting.
                ^-- SC2006: Use $(..) instead of legacy `..`.


In scfdotctl line 169:
mv hostname.pdf `hostname`.pdf
                ^-- SC2046: Quote this to prevent word splitting.
                ^-- SC2006: Use $(..) instead of legacy `..`.


In scfdotctl line 171:
echo "created `hostname`.ps and `hostname`.pdf"
              ^-- SC2006: Use $(..) instead of legacy `..`.
                                ^-- SC2006: Use $(..) instead of legacy `..`.

```

**Comment by xen0l** [comment]
@grueni also can you explain what's wrong with ```dot -Tps | ps2pdf``` or ```dot -Tps```?

**Comment by grueni** [comment]
There is nothing wrong with this form. The commands create simpler results.
I used as output format eps for the legend.dot because this file is used as encapsulated postscript.
And ps2 for hostname.dot because this creates a postscript file which is better usable for the conversion to pdf.

http://www.graphviz.org/content/output-formats#aeps
http://www.graphviz.org/content/output-formats#aps2

**Review by xen0l — APPROVED** [review]

File: components/sysutils/scfdot/files/scfdotctl [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+# Options to pass to dot when rendering the graph.  Consider increasing
+# mclimit, which dictates how long dot spends optimizing node placement.  It
+# defaults to 1.0; 100 should produce good output, but it may take a long
+# time.
+DOTOPTS=-Gmclimit=100
+
+HOSTNAME=`hostname`
+
+cat << EOF > /tmp/setpage.awk
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: Should we use mktemp or at least paid to generate unique file names for temporary files?

Comment (pyhalov) [reply components/sysutils/scfdot/files/scfdotctl]: s/paid/pid/

**Comment by xen0l** [comment]
@pyhalov are you OK with merging this?

File: components/sysutils/scfdot/files/scfdotctl [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+{ print }
+EOF
+
+
+scfdot "${SCFDOTOPTSL}"      > /tmp/legend.dot.$$
+dot -Teps /tmp/legend.dot.$$ > /tmp/legend.ps.$$
+# awk doesn't like quoted argument 
+awk -f /tmp/enlarge.awk.$$ top=3 bottom=3 \
+      /tmp/legend.ps.$$      > legend.ps
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: Why do we need to create legend.ps in $(pwd) ? 
Why it is not removed?  Why does it live in $(pwd), but not in /tmp?

File: components/sysutils/scfdot/files/scfdotctl [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+scfdot "${SCFDOTOPTSH}"      > /tmp/hostname.dot.$$
+echo "If you want to speed up the process use lower values for option Gmclimit (used value: ${DOTOPTS})"
+dot -Tps2 "${DOTOPTS}" < /tmp/hostname.dot.$$ > /tmp/hostname.ps.$$
+awk -f /tmp/setpage.awk.$$ /tmp/hostname.ps.$$ > hostname.ps
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: Again, do we need to create hostname.ps and hostname.pdf in $(pwd)?

File: components/sysutils/scfdot/files/scfdotctl (hunk 3 above)
Comment (grueni) [reply components/sysutils/scfdot/files/scfdotctl]: The 3 files ${HOSTNAME}.pdf , ${HOSTNAME}.ps, and legend.ps are the final results.
${HOSTNAME}.ps loads legend.ps as embedded postscript file.
The files in /tmp are intermediary results.

File: components/sysutils/scfdot/files/scfdotctl (hunk 4 above)
Comment (grueni) [reply components/sysutils/scfdot/files/scfdotctl]: They are created in ${pwd} and then renamed to the final names ${HOSTNAME}.pdf respective ${HOSTNAME}.ps.

Comment (pyhalov) [reply components/sysutils/scfdot/files/scfdotctl]: can we store only final files in $(pwd) ?

Comment (grueni) [reply components/sysutils/scfdot/files/scfdotctl]: no problem, I will change it

**Comment by grueni** [comment]
I stand corrected: ${HOSTNAME}-legend.ps is contained in ${HOSTNAME}.ps and can be deleted.

File: components/sysutils/scfdot/files/scfdotctl [hunk 5]
```diff
@@ -160,17 +159,14 @@ scfdot "${SCFDOTOPTSL}"      > /tmp/legend.dot.$$
 dot -Teps /tmp/legend.dot.$$ > /tmp/legend.ps.$$
 # awk doesn't like quoted argument 
 awk -f /tmp/enlarge.awk.$$ top=3 bottom=3 \
-      /tmp/legend.ps.$$      > legend.ps
+      /tmp/legend.ps.$$      > "${HOSTNAME}-legend.ps"
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: If ${HOSTNAME}-legend.ps is a temporary file, what does it do in ${PWD}?

Comment (grueni) [reply components/sysutils/scfdot/files/scfdotctl]: During testing I realized that dot has problems.
graphviz seems to be broken.
dot links to a bunch of libraries which need the egl package.
egl package is owned by x11/library/mesa.
Without egl package pkgconfig dies when used for graphic libraries which are dependent from egl.

**Comment by grueni** [comment]
This is now a simpler version of the script.
Tested with dot 2.40.1.

File: components/sysutils/scfdot/files/scfdotctl [hunk 6]
```diff
@@ -0,0 +1,174 @@
+#!/bin/sh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL)". You may
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: Misprint in license header ("CDDL)"

File: components/sysutils/scfdot/files/scfdotctl [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+# Options to pass to scfdot.  This limits the graph to 300" by 42", includes
+# legend.ps as the legend (built below), and consolidates inetd services into
+# a single node.  See the comment at the top of scfdot.c for other options.
+# Edit please directly below in the second call of scfdot.
+
+HOSTNAME=`hostname`
+SCFDOTOPTSL='-L'
+
+# Margin, in inches, to include above and below the legend.
+# LEGEND_MARGIN=3
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: Variable never used

**Comment by pyhalov** [comment]
Overall looks fine, some minor fixes needed. Don't forget to remove manifest/sample-manifest.p5m change from this PR.

**Comment by grueni** [comment]
This commit contains the updated scfdotcl. I pushed before erroneously the old scfdotctl.

**Comment by pyhalov** [comment]
Revert changes to manifest/smaple-manifest.p5m, everything else seems fine.

File: components/sysutils/scfdot/files/scfdotctl [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+dot -Teps /tmp/legend.dot.$$ > /tmp/legend.ps.$$
+awk -f /tmp/functions.awk.$$ top="${LEGEND_MARGIN}" bottom="${LEGEND_MARGIN}" f=enlarge \
+      /tmp/legend.ps.$$      > "/tmp/${HOSTNAME}-legend.ps.$$"
+scfdot -l "/tmp/${HOSTNAME}-legend.ps.$$" -s "300,42"  -x consolidate_inetd_svcs > /tmp/hostname.dot.$$
+echo "If you want to speed up the process use lower values for option Gmclimit (used value: ${DOTOPTS})"
+dot -Tps2 ${DOTOPTS} < /tmp/hostname.dot.$$ > /tmp/hostname.ps.$$
+awk -f /tmp/functions.awk.$$ f=setpage /tmp/hostname.ps.$$ > "${HOSTNAME}.ps"
+grep BoundingBox "${HOSTNAME}.ps"|tail -1
+ps2pdf -dPDFFitPage "${HOSTNAME}.ps" "${HOSTNAME}.pdf"
+rm /tmp/legend.*.$$ /tmp/hostname.*.$$ /tmp/functions.awk.$$ "/tmp/${HOSTNAME}-legend.ps.$$"
```
Comment (pyhalov) [inline components/sysutils/scfdot/files/scfdotctl]: perhaps, we should avoid using wildcards here, just explicitly name all files?

**Pull request merged**